hw/ssi/hercules_spi.c

```diff
@@ -0,0 +1,723 @@
+/*
+ */
+
+#include "qemu/osdep.h"
+#include "hw/irq.h"
+#include "hw/sysbus.h"
+#include "qemu/log.h"
+#include "qemu/main-loop.h"
+#include "qapi/error.h"
+
+#include "sysemu/dma.h"
+
+#include "hw/ssi/hercules_spi.h"
+
+#define qemu_log_bad_offset(offset) \
+    qemu_log_mask(LOG_GUEST_ERROR, "%s: Bad offset %" HWADDR_PRIx "\n", \
+                  __func__, offset);
+
+
+/*
+ *  Current assumptions in this MIBSPI implementation
+ *
+ *   - Only MIBSPI mode is implemented, no compatability mode
```

Comment:
I don't think this is true anymore. I think I had to implement compatibility mode too. Right?

Reply:
Yea, we use it on mibspi5 + dma